src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,359 @@
+	delete data.referrers.types[ 'totals' as unknown as number ];
+
+	// Returns an internationalized referrer title based on the passed key.
+	const getKeyTitle = ( key: string ): string => {
+		switch ( key ) {
+			case 'social': return __( 'Social', 'wp-parsely' );
+			case 'search': return __( 'Search', 'wp-parsely' );
+			case 'other': return __( 'Other', 'wp-parsely' );
+			case 'internal': return __( 'Internal', 'wp-parsely' );
+			case 'direct': return __( 'Direct', 'wp-parsely' );
+		}
+
+		return key;
+	};
+
+	return (
+		<div className="section referrer-types">
+			<div className="section-title">{ __( 'Referrers (Page Views)', 'wp-parsely' ) }</div>
+
+			<div className="multi-percentage-bar">{
+				Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+					const ariaLabel = sprintf(
+						/* translators: 1: Referrer type, 2: Percentage value, %%: Escaped percent sign */
+						__( '%1$s: %2$s%%', 'wp-parsely' ),
+						getKeyTitle( key ), value.viewsPercentage
+					);
+
+					return (
+						<div aria-label={ ariaLabel }
+							className={ 'bar-fill ' + key } key={ key }
+							style={ { width: value.viewsPercentage + '%' } }>
+						</div>
+					);
+				} ) }
+			</div>
+
+			<table>
+				<thead>
+					<tr>{
+						Object.keys( data.referrers.types ).map( ( key ) => {
+							return <th key={ key }>{ getKeyTitle( key ) }</th>;
+						} ) }
+					</tr>
+				</thead>
+				<tbody>
+					<tr>{
+						Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+							return <td key={ key }>{ ImpreciseNumber( value.views ) }</td>;
+						} ) }
+					</tr>
+				</tbody>
+			</table>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Top Referrers" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function TopReferrersSection( data: PostPerformanceData ) {
+	let totalViewsPercentage = 0;
+	return (
+		<div className="section top-referrers">
+			<table>
+				<thead>
+					<tr>
+						<th scope="col">{ __( 'Top Referrers', 'wp-parsely' ) }</th>
+						<th colSpan={ 2 } scope="colgroup">{ __( 'Page Views', 'wp-parsely' ) }</th>
+					</tr>
+				</thead>
+				<tbody>{
+					Object.entries( data.referrers.top ).map( ( [ key, value ] ) => {
+						if ( key === 'totals' ) {
+							totalViewsPercentage = Number( value.viewsPercentage );
+							return null;
+						}
+
+						let referrerUrl = key;
+						if ( key === 'direct' ) {
+							referrerUrl = __( 'Direct', 'wp-parsely' );
+						}
+
+						/* translators: %s: Percentage value, %%: Escaped percent sign */
+						const ariaLabel = sprintf( __( '%s%%', 'wp-parsely' ), value.viewsPercentage ); // eslint-disable-line @wordpress/valid-sprintf
+
+						return (
+							<tr key={ key }>
+								<th scope="row">{ referrerUrl }</th>
+								<td>
+									<div aria-label={ ariaLabel }
+										className="percentage-bar"
+										style={ { '--bar-fill': value.viewsPercentage + '%' } as React.CSSProperties }>
```

Comment:
Neat! I didn't know you could do this. I thought you had to render a `<style>` element for this. I guess React doesn't know how to type this though. :) 👍